hudi-client/src/main/java/org/apache/hudi/client/HoodieWriteClient.java

```diff
@@ -701,8 +704,6 @@ protected void completeCompaction(HoodieCommitMetadata metadata, JavaRDD<WriteSt
     finalizeWrite(table, compactionCommitTime, writeStats);
     LOG.info("Committing Compaction " + compactionCommitTime + ". Finished with result " + metadata);
 
-    table.metadataWriter(jsc).update(metadata, compactionCommitTime);
-
```

Comment:
Clarification: Since we are not updating the metadata table here, for compactions the metadata table will be updated the next time the HoodieWriteClient is created. 

But for commit operations the metadata table (may) be updated right after the commit completes in "postCommit" function. Right?

Reply:
yes correct. even for compaction, once the compaction commits, it will trigger a sync. I think we should ensure the sync only happens for commit operations (i.e only one allowed at the moment). I ll double check this.